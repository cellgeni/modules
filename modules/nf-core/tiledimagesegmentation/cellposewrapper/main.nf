process TILEDIMAGESEGMENTATION_CELLPOSEWRAPPER {
    tag "$meta.id"
    label 'process_medium'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        "quay.io/cellgeni/tiled_cellpose:0.1.3":
        "quay.io/cellgeni/tiled_cellpose:0.1.3"}"

    input:
    tuple val(meta), val(x_min), val(y_min), val(x_max), val(y_max), path(image), val(cell_diameter)

    output:
    tuple val(meta), path("${prefix}/${prefix}_cp_outlines.wkt"), emit: wkts
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    prefix = "${meta.id}-${x_min}_${y_min}_${x_max}_${y_max}-diam_${cell_diameter}"
    def args = task.ext.args ?: ''  
    """
    export NUMBA_CACHE_DIR=./numba_cache
    cellpose_seg.py run \
        --image ${image} \
        --x_min ${x_min} \
        --y_min ${y_min} \
        --x_max ${x_max} \
        --y_max ${y_max} \
        --cell_diameter ${cell_diameter} \
        --out_dir "${prefix}" \
        ${args}
    
    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        TILEDIMAGESEGMENTATION_CELLPOSEWRAPPER : \$(echo \$(cellpose_seg.py version))
    END_VERSIONS
    """


    stub:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}-${x_min}_${y_min}_${x_max}_${y_max}-diam_${cell_diameter}"
    """
    mkdir "${prefix}"
    touch "${prefix}/${prefix}_cp_outlines.wkt"

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        TILEDIMAGESEGMENTATION_CELLPOSEWRAPPER : \$(cellpose_seg.py version)
    END_VERSIONS
    """
}