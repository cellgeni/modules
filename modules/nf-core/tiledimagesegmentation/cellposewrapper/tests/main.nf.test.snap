{
    "mouse_heart - tiff - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test-0_0_100_100-diam_20_cp_outlines.wkt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    
                ],
                "2": [
                    "versions.yml:md5,ad4f5dcc8398626ee094320e735869f2"
                ],
                "versions": [
                    "versions.yml:md5,ad4f5dcc8398626ee094320e735869f2"
                ],
                "wkts": [
                    [
                        {
                            "id": "test"
                        },
                        "test-0_0_100_100-diam_20_cp_outlines.wkt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-30T10:12:34.322919301"
    },
    "mouse_heart - tiff - 500": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test-0_0_500_500-diam_20_cp_outlines.wkt:md5,fe837bbb6adc3d13928ed9b3405e4d9c"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "test-0_0_500_500-diam_20_cp_masks.png:md5,a1e143c85d8314e9abe9d2d35ea9760a",
                            "test-0_0_500_500-diam_20_outlines.png:md5,36b8f6235c68416d267290547010f453"
                        ]
                    ]
                ],
                "2": [
                    "versions.yml:md5,eae9d137330a6afbf958d9c23557df71"
                ],
                "versions": [
                    "versions.yml:md5,eae9d137330a6afbf958d9c23557df71"
                ],
                "wkts": [
                    [
                        {
                            "id": "test"
                        },
                        "test-0_0_500_500-diam_20_cp_outlines.wkt:md5,fe837bbb6adc3d13928ed9b3405e4d9c"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-30T10:09:18.436125763"
    },
    "mouse_heart - tiff - 200": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test-0_0_200_200-diam_20_cp_outlines.wkt:md5,b5d958738a1e7f30115b2ca3937755e4"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "test-0_0_200_200-diam_20_cp_masks.png:md5,2438cd29e8c392d7dd896c4abe763d50",
                            "test-0_0_200_200-diam_20_outlines.png:md5,ae0e50c7f2ae8af015e989f8eb8d2ff8"
                        ]
                    ]
                ],
                "2": [
                    "versions.yml:md5,eae9d137330a6afbf958d9c23557df71"
                ],
                "versions": [
                    "versions.yml:md5,eae9d137330a6afbf958d9c23557df71"
                ],
                "wkts": [
                    [
                        {
                            "id": "test"
                        },
                        "test-0_0_200_200-diam_20_cp_outlines.wkt:md5,b5d958738a1e7f30115b2ca3937755e4"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-30T10:10:44.110811691"
    },
    "mouse_heart - tiff - 20": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test-0_0_20_20-diam_20_cp_outlines.wkt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    
                ],
                "2": [
                    "versions.yml:md5,eae9d137330a6afbf958d9c23557df71"
                ],
                "versions": [
                    "versions.yml:md5,eae9d137330a6afbf958d9c23557df71"
                ],
                "wkts": [
                    [
                        {
                            "id": "test"
                        },
                        "test-0_0_20_20-diam_20_cp_outlines.wkt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-30T10:11:28.634716159"
    },
    "mouse_heart - tiff - 200 - large cell diameter": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test-0_0_200_200-diam_50_cp_outlines.wkt:md5,794454b80771bfb41532a6eb5ebf2dff"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "test-0_0_200_200-diam_50_cp_masks.png:md5,eb0c89bce655669134a7c08108a9fbe4",
                            "test-0_0_200_200-diam_50_outlines.png:md5,23c766ec26a1a6f5ec107c855ec9dbbf"
                        ]
                    ]
                ],
                "2": [
                    "versions.yml:md5,eae9d137330a6afbf958d9c23557df71"
                ],
                "versions": [
                    "versions.yml:md5,eae9d137330a6afbf958d9c23557df71"
                ],
                "wkts": [
                    [
                        {
                            "id": "test"
                        },
                        "test-0_0_200_200-diam_50_cp_outlines.wkt:md5,794454b80771bfb41532a6eb5ebf2dff"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-30T10:12:09.918190118"
    }
}